ui/utils/swap.ts

```
@@ -41,6 +43,8 @@ export type QuoteUpdate = {
   quoteRequest: SwapQuoteRequest
   timestamp: number
   amount: string
+  // Added error property to better handle and display error states
+  error?: string
```

I wonder if the correct way to model this is something like:

```typescript
type QuoteUpdate = current

type QuoteUpdateResult = QuoteUpdate | { error: string, previousQuote: QuoteUpdate }
```

Or even having the error state not include the quote at all. In particular, while I love retries, what I don't love about just having an error alongside the quote update result is that the current version leaves the quote unchanged rather than return any made-up values (e.g. below we set a 0 gas price in case of failure).

---

done!